state/store.go

```diff
@@ -465,7 +477,16 @@ func (store dbStore) PruneABCIResponses(targetRetainHeight int64) (int64, int64,
 			defer batch.Close()
 		}
 	}
-	return pruned + batchPruned, targetRetainHeight, batch.WriteSync()
+	if err = batch.WriteSync(); err != nil {
+		return pruned + batchPruned, targetRetainHeight, err
+	}
+
+	if forceCompact && store.Compact {
+		if pruned+batchPruned >= store.CompactionInterval || targetRetainHeight-lastRetainHeight >= store.CompactionInterval {
```

Comment:
I also have a problem with this condition. If understood the code well
* if condition `pruned+batchPruned >= store.CompactionInterval` is true,
* then condition `targetRetainHeight-lastRetainHeight` is also true
... because `pruned+batchPruned` will increase by 1 at every iteration of the loop.
Since both conditions are combined with a logical OR, then the second condition is redundant.

Thinking more generally, why do we care about `targetRetainHeight` and `lastRetainHeight` here? Shouldn't we just care about the first condition, i.e., the actual number of ABCI responses pruned?